CRAN notice on uninitialized variable

Incoming email by Brian Ripley says

> Dear maintainer,
> 
> Please see the problems shown on
> <https://cran.r-project.org/web/checks/check_results_RcppSMC.html>.
> 
> It has always been the intention to report
> -Winitialized warnings, but the format had changed.
> clang (LLVM and Apple) reports many more than gcc.
> 
> Please correct before 2023-04-04 to safely retain your package on CRAN.

which gets us to [CRAN results for RcppSMC](https://cran.r-project.org/web/checks/check_results_RcppSMC.html) which currently has for his Fedora machine with `clang++-16`:

> Version: 0.2.6
> Check: whether package can be installed
> Result: WARN
>    Found the following significant warnings:
>     ../inst/include/history.h:70:75: warning: field 'number' is uninitialized when used here [-Wuninitialized]
>    See ‘/data/gannet/ripley/R/packages/tests-clang/RcppSMC.Rcheck/00install.out’ for details.
>    * used C++ compiler: ‘clang version 16.0.0’
> Flavor: r-devel-linux-x86_64-fedora-clang

but it turns out that a very simple change takes care of it (in the 'tests still pass' sense) _and_ makes the warning go aways (though I used `clang++-15` here; he noted the different output format):

```diff
modified   inst/include/history.h
@@ -63,9 +63,9 @@ namespace smc {
     template <class Space>class historyelement
     {
     private:
-        long number; //!< The number of particles (presently redundant as this is not a function of iteration)
-        int nAccepted; //!< Number of MCMC moves accepted during this iteration.
-        int nRepeat; //!< Number of MCMC iterations performed at this iteration (per particle)
+        long number = 0; //!< The number of particles (presently redundant as this is not a function of iteration)
+        int nAccepted = 0; //!< Number of MCMC moves accepted during this iteration.
+        int nRepeat = 0; //!< Number of MCMC iterations performed at this iteration (per particle)
         population<Space> pop; //!< The particles themselves (values and weights)
         arma::Col<unsigned int> ancestorIndices = arma::Col<unsigned int>(number);
         historyflags flags; //!< Flags associated with this iteration.
```

so I will PR this in a moment.

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,7 @@
 2023-03-21  Dirk Eddelbuettel  <edd@debian.org>
 
+	* DESCRIPTION (Version, Date): Roll minor version
+
 	* inst/include/history.h (historyelement): Initialize three variable
 	which addresses a clang++ notice
 

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2023-03-21  Dirk Eddelbuettel  <edd@debian.org>
+
+	* inst/include/history.h (historyelement): Initialize three variable
+	which addresses a clang++ notice
+
 2022-11-29  Dirk Eddelbuettel  <edd@debian.org>
 
 	* .github/workflows/ci.yaml (jobs): Update to actions/checkout@v3

--- a/ChangeLog
+++ b/ChangeLog
@@ -5,6 +5,8 @@
 	* inst/include/history.h (historyelement): Initialize three variable
 	which addresses a clang++ notice
 
+	* man/cSMCexamples.Rd: URL update to please R CMD check
+
 2022-11-29  Dirk Eddelbuettel  <edd@debian.org>
 
 	* .github/workflows/ci.yaml (jobs): Update to actions/checkout@v3

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,8 +1,8 @@
 Package: RcppSMC
 Type: Package
 Title: Rcpp Bindings for Sequential Monte Carlo
-Version: 0.2.6.1
-Date: 2022-04-26
+Version: 0.2.6.2
+Date: 2023-03-21
 Author: Dirk Eddelbuettel, Adam M. Johansen, Leah F. South and Ilya Zarubin
 Maintainer: Dirk Eddelbuettel <edd@debian.org>
 Description: R access to the Sequential Monte Carlo Template Classes

--- a/inst/include/history.h
+++ b/inst/include/history.h
@@ -63,9 +63,9 @@ namespace smc {
     template <class Space>class historyelement
     {
     private:
-        long number; //!< The number of particles (presently redundant as this is not a function of iteration)
-        int nAccepted; //!< Number of MCMC moves accepted during this iteration.
-        int nRepeat; //!< Number of MCMC iterations performed at this iteration (per particle)
+        long number = 0; //!< The number of particles (presently redundant as this is not a function of iteration)
+        int nAccepted = 0; //!< Number of MCMC moves accepted during this iteration.
+        int nRepeat = 0; //!< Number of MCMC iterations performed at this iteration (per particle)
         population<Space> pop; //!< The particles themselves (values and weights)
         arma::Col<unsigned int> ancestorIndices = arma::Col<unsigned int>(number);
         historyflags flags; //!< Flags associated with this iteration.

--- a/man/cSMCexamples.Rd
+++ b/man/cSMCexamples.Rd
@@ -122,10 +122,10 @@
 \references{
   A. M. Johansen. SMCTC: Sequential Monte Carlo in C++.
   Journal of Statistical Software, 30(6):1-41, April
-  2009. \url{https://www.jstatsoft.org/v30/i06/paper}
+  2009. \url{https://www.jstatsoft.org/article/view/v030i06}
 }
 
-\seealso{The SMCTC paper and code at \url{https://www.jstatsoft.org/v30/i06/paper}.}
+\seealso{The SMCTC paper and code at \url{https://www.jstatsoft.org/article/view/v030i06}.}
 
 \author{Adam M. Johansen, Dirk Eddelbuettel, Leah South and Ilya Zarubin}
 \keyword{programming}

